resource/name_test.go

```diff
@@ -0,0 +1,165 @@
+package resource
+
+import (
+	"testing"
+
+	"go.viam.com/test"
+)
+
+func TestNewName(t *testing.T) {
+	type testCase struct {
+		api          API
+		nameString   string
+		expectedName Name
+	}
+
+	camAPI := API{Type: APIType{Namespace: APINamespace("rdk"), Name: "component"}, SubtypeName: "camera"}
+	test.That(t, camAPI, test.ShouldResemble, APINamespaceRDK.WithComponentType("camera"))
+
+	motionAPI := API{Type: APIType{Namespace: APINamespace("rdk"), Name: "service"}, SubtypeName: "motion"}
+	test.That(t, motionAPI, test.ShouldResemble, APINamespaceRDK.WithServiceType("motion"))
+
+	tcs := []testCase{
+		{
+			api:          camAPI,
+			nameString:   "cam-1",
+			expectedName: Name{API: camAPI, Remote: "", Name: "cam-1"},
+		},
+		{
+			api:        camAPI,
+			nameString: "remote:cam-1",
+
+			expectedName: Name{API: camAPI, Remote: "remote", Name: "cam-1"},
+		},
+		{
+			api:        camAPI,
+			nameString: "remoteA:remoteB:cam-1",
+
+			expectedName: Name{API: camAPI, Remote: "remoteA:remoteB", Name: "cam-1"},
+		},
+		{
+			api:        motionAPI,
+			nameString: "builtin",
+
+			expectedName: Name{API: motionAPI, Remote: "", Name: "builtin"},
+		},
+		{
+			api:        motionAPI,
+			nameString: "remote:builtin",
```

Comment:
can we add a non-builtin non-camera test as weel, just for completeness since the resource graph special cases the motion builtin.